edb/pgsql/dbops/types.py

```diff
@@ -59,6 +59,28 @@ def code(self, block: base.PLBlock) -> str:
         ''')
 
 
+def type_oid(name):
```

Comment:
Will this query be run for every type?  Maybe read them all in one query?

Reply:
This query is ran for every `CREATE TYPE` DDL, to *write* the newly added type's OID.  This isn't really performance-critical.  The *reading* of OIDs for the newly added types is done by a single query all at once.